utils/src/server/rpc_module.rs

```diff
@@ -242,8 +246,45 @@ impl SubscriptionSink {
 				params: JsonRpcNotificationParams { subscription: *sub_id, result: &*result },
 			})?;
 
-			// Track broken connections
-			if sender.unbounded_send(msg).is_err() {
+			// Log broken connections
+			if sender.sink.unbounded_send(msg).is_err() {
+				errored.push((*conn_id, *sub_id));
+			}
+		}
+
+		// Remove broken connections
+		for entry in errored {
+			subs.remove(&entry);
+		}
+
+		Ok(())
+	}
+
+	/// Send a message to all subscriptions that could parse `P` as input.
+	///
+	/// F: is closure that you need to provide to apply on the input P.
+	pub fn send_all_with_params<T, F>(&self, f: F) -> anyhow::Result<()>
+	where
+		F: Fn(&P) -> T,
```

Comment:
do we need `F: Fn(&mut P) -> T` too?

i,e, to mutate the param?

Reply:
Hmm. Maybe, but until we know we do I say we wait.

Reply:
it looks like I need it for my `author_WatchExtrinsic` branch, I will try it out now

Thus, changed to `F: Fn(&mut Params) -> T`

Reply:
ofc, I found an edge-case where this doesn't work,

old code:

```rust
loop {
	let handle = match self.xt_sink.next() {
		Some(h) => h,
		None => continue,
	};
	
	let xt = handle.params();
	let best_block_hash = self.client.info().best_hash;
	let dxt = TransactionFor::<P>::decode(&mut &xt[..]).unwrap();
	let stream = match self.pool.submit_and_watch(&generic::BlockId::hash(best_block_hash), TX_SOURCE, dxt).await {
		Ok(stream) => stream,
		// just drop the subscription.
		Err(_) => continue,
	};

	// TODO: ugly blocks the executor from accepting other subs.
	stream.for_each(|item| {
		let _ = handle.send_all(&item);
		futures::future::ready(())
	}).await;
}
```

Basically, I found no way of porting this design because firstly, the param(s) is fetched to create a stream and that stream is polled and when a new element is received that is send by the sink. Thus, I don't know how to do it without having an API for fetch params and send it out only on that subscriber.

I suppose it could be possible to wrap the stream in the closure to seems quite messy to poll in and so on with having some other API to do it.

Reply:
hmm, do you have a branch?

Reply:
https://github.com/paritytech/substrate/tree/na-jsonrpsee-params-subscribe haven't pushed the stuff on it yet